src/openzeppelin/upgrades/library.cairo

```diff
@@ -52,47 +51,37 @@ namespace Proxy:
         end
 
         Proxy_initialized.write(TRUE)
-        Proxy_admin.write(proxy_admin)
+        _set_admin(proxy_admin)
```

Comment:
should we emit `AdminChanged` on initialization?

Reply:
I assumed we should, it emits in `_set_admin`

Reply:
I know that's why i was asking. What's the solidity default behavior?

Reply:
Ah gotcha. The Solidity behavior does include the event upon construction (with the `TransparentUpgradeableProxy`):
- ERC1967Upgrade `_changeAdmin` with event: [https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Upgrade.sol#L133-L136](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Upgrade.sol#L133-L136)
- TUP contract `constructor` invoking `_changeAdmin`: [https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/transparent/TransparentUpgradeableProxy.sol#L34-L40](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/transparent/TransparentUpgradeableProxy.sol#L34-L40)